packages/react/src/IDE/IDE.tsx

```diff
@@ -353,234 +337,188 @@ const _Editor = memo(
-              }, fixedDelay)
-            } else if (pre.getAttribute('data-has-suggestion') === 'true' && fixedDelay > 0) {
-              setTimeout(() => {
-                pre.classList.add(animationStyles['Animation--active'])
-              }, fixedDelay)
-              delay = fixedDelay
-            } else {
-              setTimeout(() => {
-                pre.classList.add(animationStyles['Animation--active'])
-              }, 200 * index)
-              delay = 200 * index
-            }
-            index++
+            delay = copilotSuggestionDelay
           }
-          totalDelay += delay
-        }
-        setAnimationIsActive(false)
 
-        setTimeout(() => {
-          rootRef.current?.setAttribute('data-animation-active', 'false')
-          // reenable button
-          if (showReplayButton) {
-            if (buttonRef.current) {
-              buttonRef.current.disabled = false
-            }
-          }
-          // reenable other tabs
-          if (tabsRef.current) {
-            const tabs = tabsRef.current.querySelectorAll('button')
-            for (const tab of Array.from(tabs)) {
-              tab.disabled = false
-            }
-          }
-        }, totalDelay)
+          const timeout = setTimeout(() => {
+            pre.classList.add(animationStyles['Animation--active'])
+          }, delay)
 
-        return () => {
-          if (pres) {
-            for (const pre of Array.from(pres)) {
-              pre.classList.remove(animationStyles['Animation--active'])
-            }
-          }
+          setTimeouts(prev => [...prev, timeout])
+
+          delay += 200
         }
-      }, [
-        activeFile,
-        activeTab,
-        triggerAnimation,
-        animationIsActive,
-        localAnimationCouner,
-        buttonRef,
-        showReplayButton,
-        rootRef,
-      ])
+
+        const animationEndTimeout = setTimeout(() => {
+          setHasAnimated(true)
+          setIsAnimating(false)
+        }, delay)
+
+        setTimeouts(prev => [...prev, animationEndTimeout])
+      }, [hasAnimated, isAnimating])
+
+      const onTabActivate = useCallback(() => {
+        resetAnimation()
+      }, [resetAnimation])
+
+      const tabs = useTabs({
+        defaultTab: activeTab.toString(),
+        autoActivate: true,
+        onTabActivate,
+      })
+
+      useEffect(() => {
+        runAnimation()
+      }, [runAnimation, triggerAnimation])
 
       return (
         <div
           className={clsx(styles.IDE__Editor, styles[`IDE__Editor--${size}`])}
-          ref={rootRef}
+          ref={ref}
           data-testid={testId || testIds.editor}
           {...rest}
         >
-          <div className={styles['IDE__Editor-tabs']} ref={tabsRef} data-testid={testIds.editorTabs}>
-            {files.map((file, index) => {
+          <div className={styles['IDE__Editor-tabs']} data-testid={testIds.editorTabs} {...tabs.getTabListProps()}>
+            {files.map((file, i) => {
```

Comment:
Nit and just curious, was there a reason for removing the verbose name for index here?

Reply:
No there wasn't, and I hadn't realised I'd done it to be honest. I refactored around there a couple of times and that must have just slipped through. If you'd prefer `index` I can update it?

Reply:
Updated anyway 🙂

f6b1b2a44d81ab9499d07bd0480c34319f238e68